### DISCUSSION TITLE
How to switch on and off a different configuration of raw

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#2959] [Q&A] [answered]
I am a newbie user of `typst`. Consider the document below.

How can we easily switch on and off the configuration of `raw`, so that we can use a different configuration when `raw` is inline and when we want `raw` to be a block.

One approach would be to be redefining constantly the `raw` element, but that is not very convenient.

````
#show raw: it => block(
  fill: luma(240),
  width: 100%,
  inset: 8pt,
  radius: 5pt,
  text(it)
)

```python
x = 4 * 3
y = x ** 2

def f(x):
    return x
```

Now, I want to use raw inline: `python`.
````

**Comment by PgBiel** [comment, marked as answer]
Use [.where](https://typst.app/docs/reference/foundations/function/#definitions-where):

```typ
#show raw.where(block: true): it => block(
  fill: luma(240),
  width: 100%,
  inset: 8pt,
  radius: 5pt,
  text(it)
)
```

Note that `text(it)` is the same as `it` when `it` is already content (which is the case for a `raw` block), so you can replace `text(it)` by `it`. But even that can be simplified as follows:

```typ
#show raw.where(block: true): block.with(
  fill: luma(240),
  width: 100%,
  inset: 8pt,
  radius: 5pt
)
```

**Reply by psads-git** [reply]
Thanks a lot, @PgBiel : that is what I was looking for!

**Reply by PgBiel** [reply]
No problem! By the way, this discussion might be better suited in the Q&A category.

**Reply by psads-git** [reply]
I have just transferred this discussion to Q&A category, @PgBiel. Thanks!

**Reply by PgBiel** [reply]
Great! You're welcome.
